prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1730,7 +1730,8 @@ static size_t rxr_ep_post_rts(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_ent
 
 	tx_entry->bytes_sent += data_sent;
 
-	if (!(efa_mr_cache_enable && tx_entry->total_len > data_sent))
+	if ((rxr_env.enable_shm_transfer && peer->is_local) ||
```

Comment:
I must have missed this commit in the internal review.

This change works for the subsequent data packets. Are we doing the right thing for the RTS payload when sending over shm?

Reply:
shm does not require local memory registration, this patch skip this

Reply:
Yep, I was asking more in the context of not copying/registering payload for shm when the cache is enabled, but I checked we do the right thing. We'll reuse the app buffer for the RTS payload with a change I will be making soon.